### PR TITLE
OEUI-91: Implement axios instance in all previously implemented actions

### PR DESCRIPTION
**Pull request by geofrocker**
## JIRA TICKET NAME:
[OEUI-91: Implement axios instance in all previously implemented actions](https://issues.openmrs.org/browse/OEUI-91)
### SUMMARY:
This ticket will require the creation of a config file that will create an axios instance and set the baseURL. The task would also require changing all the previous api calls to use the axios instance

### CODE REVIEW

File: app/js/actions/activeOrderAction.js [hunk 1]
```diff
@@ -18,8 +17,8 @@ const activeOrderActionError = error => ({
 });
 
 const activeOrderAction = (careSetting, patientUuid) => (dispatch) => {
-  dispatch(loading('FETCH_ACTIVE_ORDER', true));
-  return axios.get(`${apiBaseUrl}/order?careSetting=${careSetting}&patient=${patientUuid}&t=drugorder&v=full`, {
+  dispatch(loading(true));
```
Comment (fuchodeveloper) [inline app/js/actions/activeOrderAction.js]: Any reason for replacing the existing code on line 20?

File: app/js/config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
+
+export const http = axios.create({
+  baseURL: apiBaseUrl,
+  headers: {
+    accept: 'application/json',
+  },
+});
+
+export default http;
```
[the remark this answers is not in the document]
Comment (geofrocker) [reply app/js/config.js]: Thanks @fuchodeveloper. Fixed

File: app/js/actions/activeOrderAction.js (hunk 1 above)
Comment (geofrocker) [reply app/js/actions/activeOrderAction.js]: I did not intend to remove it. Have added it back, thank u

**Comment by geofrocker** [comment]
@kwahalf

File: app/js/actions/addDrugOrder.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 import networkError from './networkError';
 import loading from './loading';
 
-const contextPath = window.location.href.split('/')[3];
-const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
-
+export const postDrugOrderLoading = status => ({
+  type: POST_DRUG_ORDER_LOADING,
+  status,
+});
```
Comment (larrystone) [inline app/js/actions/addDrugOrder.js]: I guess lines 10 -13 are no more needed

File: app/js/actions/careSetting.js [hunk 4]
```diff
@@ -26,7 +23,7 @@ const fetchPatientCareSettingLoading = status => ({
 
 const fetchPatientCareSetting = () => (dispatch) => {
   dispatch(fetchPatientCareSettingLoading(true));
```
Comment (larrystone) [inline app/js/actions/careSetting.js]: I guess we agreed to replace our loader actions with a generic one?

Comment (geofrocker) [reply app/js/actions/careSetting.js]: The `Integrate a generic loader` ticket is still in progress and I believe you concerns are being worked on by @fuchodeveloper

File: app/js/actions/addDrugOrder.js (hunk 3 above)
Comment (geofrocker) [reply app/js/actions/addDrugOrder.js]: This is being worked on in another ticket [here](https://issues.openmrs.org/browse/OEUI-84) by @fuchodeveloper  and is not part of this task

File: app/js/actions/careSetting.js (hunk 4 above)
Comment (larrystone) [reply app/js/actions/careSetting.js]: Okay then

File: app/js/actions/addDrugOrder.js (hunk 3 above)
Comment (larrystone) [reply app/js/actions/addDrugOrder.js]: Alright

File: app/js/actions/careSetting.js (hunk 4 above)
Comment (larrystone) [reply app/js/actions/careSetting.js]: Okay then, nice work

File: app/js/actions/addDrugOrder.js (hunk 3 above)
Comment (fuchodeveloper) [reply app/js/actions/addDrugOrder.js]: Currently working on this

File: app/js/config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (fuchodeveloper) [reply app/js/config.js]: All good

File: app/js/actions/addDrugOrder.js (hunk 3 above)
Comment (larrystone) [reply app/js/actions/addDrugOrder.js]: Okay

File: app/js/config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (geofrocker) [reply app/js/config.js]: Thank you @Betty-Kebenei Have updated it

File: app/js/config.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import axios from 'axios';
+
+const contextPath = window.location.href.split('/')[3];
+const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
+
+export const axiosInstance = axios.create({
+  baseURL: apiBaseUrl,
+  headers: {
+    accept: 'application/json',
```
Comment (dkayiwa) [inline app/js/config.js]: Do we need the last commas?

Comment (geofrocker) [reply app/js/config.js]: It’s useful for consistency across multiple lines and it's actually added by the linter itself @dkayiwa

Comment (geofrocker) [reply app/js/config.js]: More explanation can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas)

Comment (dkayiwa) [reply app/js/config.js]: Super cool! Thanks 😊

**Pull request merged**